src/main/java/seedu/address/logic/parser/UntagCommandParser.java

```diff
@@ -15,42 +20,66 @@
  */
 public class UntagCommandParser implements Parser<UntagCommand> {
     public static final int MAX_LENGTH = 50;
+    public static final int MAX_INDEXES = 10;
 
     /**
-     * Parses the given {@code String} of arguments in the context of the TagCommand
-     * and returns a TagCommand object for execution.
+     * Parses the given {@code String} of arguments in the context of the UntagCommand
+     * and returns a UntagCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
     public UntagCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_TAG);
 
-        Index index;
-
-        try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
-        } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, UntagCommand.MESSAGE_USAGE), pe);
-        }
-
-        if (!argMultimap.getValue(PREFIX_TAG).isPresent()) {
+        Set<Index> indexSet = new HashSet<>();
+        Set<Tag> tagSet = new HashSet<>();
+        List<String> tagStrings = argMultimap.getAllValues(PREFIX_TAG);
+        if (tagStrings.isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, UntagCommand.MESSAGE_USAGE));
         }
 
-        String trimmedTagName = argMultimap.getValue(PREFIX_TAG).get().trim().toLowerCase();
-        boolean isEmpty = trimmedTagName.isEmpty();
-        boolean isTooLong = trimmedTagName.length() > MAX_LENGTH;
+        try {
+            List<String> indexStrings = List.of(argMultimap.getPreamble().trim().split("\\s+"));
+            if (indexStrings.size() == 1 && indexStrings.get(0).isEmpty()) {
```

Comment:
Maybe some extra variables can be added here to make the logic clearer (ie. what does 1 represent and what does indexStrings.get(0) represent?)